src/main/java/classify/textfilecode/TextFileParser.java

```diff
@@ -0,0 +1,199 @@
+package classify.textfilecode;
+
+import classify.commands.AddStudent;
+import classify.student.Student;
+import classify.student.SubjectGrade;
+import classify.ui.UI;
+import classify.user.InputParsing;
+import classify.user.InvalidCharacterException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+//@@author ParthGandhiNUS
+public class TextFileParser {
+    private static final String MAIN_REGEX = "~~";
+    private static final String FILE_SELECTION_PROMPT = "Please enter the exact name of the file " +
+            "you'd like to process:";
+    private static final String REQUEST_TO_TRY_AGAIN = "Please try again!";
+    private static final String TEXT_TYPE = "txt";
+    private static final Character DOT = '.';
+    private static final String NO_MATCH_FOUND = "No Match Found!";
+    private static final String ERROR_ACCESSING_THE_FILE = "Something went wrong while accessing the file";
+    private static final Integer NAME = 0;
+    private static final Integer GRADE_FOR_SUBJECT = 1;
+    private static final String INPUT_TEXT_FILE_DIRECTORY = "./data/inputFolder";
+    private static final String INVALID_CHAR_MESSAGE = "Invalid character found.";
+    private static final String INVALID_NUMBER_MESSAGE = "Invalid Number format found.";
+    private static final String SUBJECT_REGEX = "Subject:";
+    private static final String CLASSES_ATTENDED_REGEX = "Classes Attended:";
```

Comment:
Nitpicking again, but consider refactoring more UI-related methods out of the class for cleaner code